app/controllers/media_objects_controller.rb

```diff
@@ -296,27 +297,26 @@ def update_status
     success_count = 0
     Array(params[:id]).each do |id|
       media_object = MediaObject.find(id)
-      if cannot? :update, media_object
-        errors += ["#{media_object.title} (#{id}) (permission denied)."]
+
+      if (status == 'publish') && can?(:update, media_object)
+        media_object.publish!(user_key)
+        # additional save to set permalink
+        media_object.save(validate: false)
+        success_count += 1
+      elsif (status == 'unpublish') and can?(:unpublish, media_object)
```

Comment:
Yeah, I hate `and` (I've been bit by the precedence with the equals sign, which I think they stole from perl). No clue why I have `and` here, will remove.